### DISCUSSION TITLE
Simplifying jupyter notebook flags

### DISCUSSION THREAD

**Discussion by kolibril13** [ManimCommunity/manim#2783] [Suggestions and Proposals]
Currently, a cell looks like this:
```py
%%manim Example1 -pql -s
class Example1(Scene):
    def construct(self, **kwargs):
        self.add(Dot())
        self.wait(4)
```
but as jupyter cells most likely will not contain more than one manim Scene, it would be really convenient to not mention the scene in the cell magic, e.g. like this:
```py
%%manim -pql -s 
class Example1(Scene):
    def construct(self, **kwargs):
        self.add(Dot())
        self.wait(4)
```
My first idea would be to search the cell with a regular expression to search for `class XX(Scene)`
and fetching XX as the scene name that needs to be rendered.
However, I did not find a way yet to grab the text of the current jupyter cell, and this seems to be a very hacky solution. Any other ideas? cc @behackl

**Comment by DJMcNab** [comment]
I think the cell contents is passed in as a string to

https://github.com/ManimCommunity/manim/blob/4716331ab45692245fa91269e39c766a457a87dc/manim/utils/ipython_magic.py#L34

So I think you'd only need to add the check to that, and change https://github.com/ManimCommunity/manim/blob/4716331ab45692245fa91269e39c766a457a87dc/manim/utils/ipython_magic.py#L81 to use the found name

**Comment by kolibril13** [comment]
Nice!
I will try this.
There will be two special cases:
1. The case when there are multiple scenes.
In this case, I would throw a warning saying:
"Multiple Scenes were found in this cell, but only the first scene will be rendered."
2. that scene names are commented out.
But I think this is quite easy to ignore when all lines of the cell that start with # are skipped.

**Comment by kolibril13** [comment]
ok, here is now the implementation:
```py
string="""
class Pendulum1(ZoomedScene):
    def construct(self):
        pendulum = Pendulum(weight=1, amplitude=12, acceleration=10, length=3, time=0)
        self.add(pendulum)
        
#class Pendulum2(Scene):
#    def construct(self):
#        pendulum = Pendulum(weight=1, amplitude=12, acceleration=10, length=3, time=0)
#        self.add(pendulum)
"""
```
```py
scenes= []
for li in string.splitlines():
    if not li.startswith("#"):
        result =  re.search(r'class(.*?)\((SceneGraph|Scene|ThreeDScene|ZoomedScene|MovingCameraScene|ZoomedScene)', li)
        if result != None:
            scenes.append(result.group(1))
print(scenes[0])
```

`Pendulum1`

**Comment by behackl** [comment]
I've already mentioned this in private communications, but I'd like to have this on the issue as well: I don't like the approach where string parsing is used to detect the class name.

I'd much rather use the existing logic from `module_ops.py` which allows to call `manim file.py` and detects appropriate classes in there programatically, without string parsing.
